src/main/java/com/laytonsmith/core/constructs/CClassType.java

```diff
@@ -266,7 +266,7 @@ private CClassType(FullyQualifiedClassName type, Target t, boolean newDefinition
 	 * do that in the constructor, we need to lazy load it. We do take pains in the constructor to ensure that there is
 	 * at least no way this will throw a ClassCastException, so given that, we are able to supress that exception here.
 	 */
-	private void instantiateInvalidType(Environment env) {
+	private synchronized void instantiateInvalidType(Environment env) {
```

Comment:
This is too heavy. Use a double locking paradigm:

```java
if(invalidType == UNINITIALIZED) {
    synchronized(this) {
        if(invalidType == UNININTIALIZED) {
            // Do method body
        }
    }
}
```
This makes reads not synchronized, except the very first time this is called. But after that, it'll be non-blocking.

Reply:
Updated, and constructed the Mixed[] in a local variable to disallow a Mixed[1] with a non-initialized first element from being returned and still causing the same problem.